lib/schema/src/validate.rs

```diff
@@ -751,3 +904,304 @@ impl RegisterBlock {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{Register, RegisterBlock, RegisterField, RegisterType};
+
+    #[test]
+    fn test_extract_array() {
+        let block = RegisterBlock {
+            name: "block".into(),
+            registers: vec![
+                Register {
+                    name: "unrelated0".into(),
+                    offset: 0,
+                    ..Default::default()
+                }
+                .into(),
+                Register {
+                    name: "data0".into(),
+                    offset: 16,
+                    ..Default::default()
+                }
+                .into(),
+                Register {
+                    name: "data1".into(),
+                    offset: 20,
+                    ..Default::default()
+                }
+                .into(),
+                Register {
+                    name: "data2".into(),
+                    offset: 24,
+                    ..Default::default()
+                }
+                .into(),
+            ],
+            ..Default::default()
+        };
+        let mut block = block.validate_and_dedup().unwrap();
+        block.extract_array("data").unwrap();
+        assert_eq!(
+            block.block,
+            RegisterBlock {
+                name: "block".into(),
+                registers: vec![
+                    Register {
+                        name: "unrelated0".into(),
+                        offset: 0,
+                        ..Default::default()
+                    }
+                    .into(),
+                    Register {
+                        name: "data".into(),
+                        offset: 16,
+                        array_dimensions: vec![3,],
+                        ..Default::default()
+                    }
+                    .into(),
+                ],
+                ..Default::default()
+            }
+        );
+    }
+
+    #[test]
+    fn test_extract_array_multidimensional() {
+        let block = RegisterBlock {
+            name: "block".into(),
+            registers: vec![
+                Register {
+                    name: "unrelated0".into(),
+                    offset: 0,
+                    ..Default::default()
+                }
+                .into(),
+                Register {
+                    name: "data0".into(),
+                    offset: 16,
+                    array_dimensions: vec![16],
+                    ..Default::default()
+                }
+                .into(),
+                Register {
+                    name: "data1".into(),
+                    offset: 20,
```

Comment:
Shouldn't this offset be `16 + 4*16 = 80`?

Reply:
Ooh, nice catch! That's a bug

Reply:
Fixed.